packages/mobile/src/components/drawers/OfflineListeningDrawer.tsx

```diff
@@ -0,0 +1,217 @@
+import type { ComponentType } from 'react'
+import { useCallback, useState } from 'react'
+
+import { View } from 'react-native'
+import type { SvgProps } from 'react-native-svg'
+import { useDispatch } from 'react-redux'
+
+import IconCart from 'app/assets/images/iconCart.svg'
+import IconDownload from 'app/assets/images/iconDownloadQueued.svg'
+import IconFavorite from 'app/assets/images/iconFavorite.svg'
+import IconRepost from 'app/assets/images/iconRepost.svg'
+import { Button, Switch, Text } from 'app/components/core'
+import { useDrawer } from 'app/hooks/useDrawer'
+import { setVisibility } from 'app/store/drawers/slice'
+import { requestDownloadAllFavorites } from 'app/store/offline-downloads/slice'
+import { makeStyles } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+import { NativeDrawer } from '../drawer'
+
+const useDrawerStyles = makeStyles(({ spacing, palette, typography }) => ({
+  container: {
+    paddingTop: spacing(4),
+    paddingBottom: spacing(10),
+    flexDirection: 'column',
+    paddingHorizontal: spacing(4),
+    rowGap: spacing(6),
+    alignItems: 'center'
+  },
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: spacing(2)
+  },
+  titleText: {
+    textTransform: 'uppercase',
+    marginTop: spacing(4)
+  },
+  descriptionText: {
+    textAlign: 'center',
+    lineHeight: typography.fontSize.large * 1.3
+  },
+  descriptionContainer: {
+    width: '100%'
+  },
+  titleIcon: {
+    position: 'relative',
```

Comment:
not sure of exact styling, but seems a bit weird that we are positioning the icon like this instead of the icon and title just being in a flex box?